components/FavoriteButton/index.js

```diff
@@ -0,0 +1,62 @@
+import React, { useState } from "react";
+import styled from "styled-components";
+import { RiStarSmileLine } from "react-icons/ri";
+import { RiStarSmileFill } from "react-icons/ri";
+import { useContext } from "react";
+import { FavoriteContext } from "../../pages/_app";
+
+export default function FavoriteButton(props) {
+  const favoritecontext = useContext(FavoriteContext);
+  const favoriteButtonHandler = () => {
+    favoritecontext.handleFavorite(props.productcard);
+  };
+
+  const [isFavorite, setIsFavorite] = useState(false);
+
```

Comment:
i saw that the favorite button on the homepage and the one on the favorite page are not connected i think it is because this state is not global.